src/Adapters/SmtpAdapter.php

```diff
@@ -4,68 +4,80 @@
 
 use Illuminate\Support\Arr;
 use Sendportal\Base\Services\Messages\MessageTrackingOptions;
-use Swift_Mailer;
-use Swift_Message;
-use Swift_SmtpTransport;
-use Swift_TransportException;
+
+use Symfony\Component\Mailer\Transport;
+use Symfony\Component\Mailer\Transport\Dsn;
+use Symfony\Component\Mailer\Transport\Smtp\EsmtpTransport;
+use Symfony\Component\Mailer\Transport\Smtp\EsmtpTransportFactory;
+use Symfony\Component\Mailer\Mailer;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mailer\Exception\TransportException;
 
 class SmtpAdapter extends BaseMailAdapter
 {
-    /** @var Swift_Mailer */
+    /** @var Mailer */
     protected $client;
 
-    /** @var Swift_SmtpTransport */
+    /** @var ESmtpTransport */
     protected $transport;
 
     public function send(string $fromEmail, string $fromName, string $toEmail, string $subject, MessageTrackingOptions $trackingOptions, string $content): string
     {
-        $failedRecipients = [];
-
+        // $failedRecipients = [];
         try {
-            $result = $this->resolveClient()->send($this->resolveMessage($subject, $content, $fromEmail, $fromName, $toEmail), $failedRecipients);
-        } catch (Swift_TransportException $e) {
+            $result = $this->resolveClient()->send($this->resolveMessage($subject, $content, $fromEmail, $fromName, $toEmail));
+        } catch (TransportException $e) {
             return $this->resolveMessageId(0);
         }
-
         return $this->resolveMessageId($result);
     }
 
-    protected function resolveClient(): Swift_Mailer
+    protected function resolveClient(): Mailer
     {
         if ($this->client) {
             return $this->client;
         }
 
-        $this->client = new Swift_Mailer($this->resolveTransport());
+        $this->client = new Mailer($this->resolveTransport());
 
         return $this->client;
     }
 
-    protected function resolveTransport(): Swift_SmtpTransport
+    protected function resolveTransport(): EsmtpTransport
     {
         if ($this->transport) {
             return $this->transport;
         }
 
-        $this->transport = new Swift_SmtpTransport(
+        $factory = new EsmtpTransportFactory();
+
+        $encryption = Arr::get($this->config, 'encryption');
+    
+        $scheme = !is_null($encryption) && $encryption === 'tls'
```

Comment:
I kinda feel like this should just be exposed in the config instead of some behind the scenes magic. Thoughts?

Reply:
encryption isn't an explicit config in symfony, so the examples I saw where checking for tls and the port to determine the schema, then mailer is configured only by a dsn string...